### PR TITLE
Created users table & script to run migrations

### PR DESCRIPTION
**Pull request by mpolisch**
## Summary
Created users table migration and a reusable init_db.ts script that scans and runs all SQL files in /migrations

## Approach & Design Decisions

- Raw SQL migrations instead of ORM because the schema is small and explicit SQL is more easily readable for the demo
- Migration script sorts files by filename, meaning the files will run in chronological order (since they're numbered 001, 002, etc.)
- IF NOT EXISTS makes the script safe to run multiple times
- TIMESTAMPTZ instead of TIMESTAMP to store timezone, removing possible ambiguity
- set_updated_at() trigger handles updated_at in pg instead of the app

## Security Considerations
- DATABASE_URL - loaded from .env with dotenv package

## Testing
**What's covered:**
- [ ] Unit tests
- [x] Edge cases (list below)

**Edge cases considered:**
- Migration run twice - IF NOT EXISTS prevents issues due to this
- Missing DATABASE_URL - init_db.ts will throw a console.error and exit the process
- Migration failure mid-run: SQL table creation wrapped with BEGIN and COMMIT to ensure ACID operations, init_db.ts will run process.exit(1) and prevent any more migrations from occurring

## Checklist
- [x] No secrets or credentials in code
- [x] Input validation in place
- [x] Error handling covers failure paths
- [x] Code is self-documenting or commented where non-obvious
- [x] No dead code or debug logs left in

## Notes
- init_db.ts is intentionally simple (no migrations table to track what's been run). In a larger application, a proper migration tool could be used to prevent re-running applied migrations

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
         "dev": "tsx watch src/index.ts",
         "build": "tsc",
         "start": "node dist/index.js",
+        "migrate": "tsx scripts/init_db.ts",
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply package.json:10]: This is more of a 'nice to have' and will be noted in the readme as a possible future enhancement

File: migrations/001_create_users.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
+BEGIN;
+
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply migrations/001_create_users.sql:2]: Not a concern, all PG instances that will be used are v13+

File: migrations/001_create_users.sql [hunk 3]
```diff
@@ -0,0 +1,23 @@
+BEGIN;
+
+CREATE TABLE IF NOT EXISTS users (
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    email VARCHAR(255) UNIQUE NOT NULL,
+    password_hash VARCHAR(255) NOT NULL,
+    created_at TIMESTAMPTZ DEFAULT NOW(),
+    updated_at TIMESTAMPTZ DEFAULT NOW()
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply migrations/001_create_users.sql]: fixed this issue.
